elasticsearch-client/lib/elasticsearch-client.test.ts

```diff
@@ -89,7 +89,7 @@ describe("ElasticsearchClient",() => {
         const result = sut.request('GET', '/thing');
 
         // THEN
-        expect(result).rejects.toEqual({ statusCode: 500 });
+        await expect(result).rejects.toEqual({ statusCode: 500 });
```

Comment:
Good catch.  FWIW, you can selectively use jest.assertions(n) as a sanity check that no promises aren't being ignored.